akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -2079,89 +2079,98 @@ private[stream] object Collect {
 /**
  * INTERNAL API
  */
-@InternalApi final private[akka] class LazyFlow[I, O, M](flowFactory: I => Future[Flow[I, O, M]])
-    extends GraphStageWithMaterializedValue[FlowShape[I, O], Future[Option[M]]] {
-  val in = Inlet[I]("lazyFlow.in")
-  val out = Outlet[O]("lazyFlow.out")
+@InternalApi private[akka] final class LazyFlow[I, O, M](flowFactory: I => Future[Flow[I, O, M]])
```

Comment:
Could we remove this implementation and replace it with the recommended `prefixAndTail` construction directly in the API?

Reply:
I'll defer that to a separate issue, this is the current underlying impl for all the lazy flow operators. It's completely internal though so fine to do later, the important thing for the RC is to get the API right.

Reply:
Created issue https://github.com/akka/akka/issues/28016 to track that